docs/proposals/1199-inferencemodel-api-evolution/README.md

```diff
@@ -0,0 +1,170 @@
+
+This proposal is intended to act as the plan of record for solution that will be implemented.
+
+## Implementation Phases
+
+### Phase 1 - Rename, Split, & Modify InferenceModel 
+A few points were used in composing justification & structure of this change:
+ - the Criticality field of InferenceModel is in use, & provides functionality 
+ - InferenceModel is an Alpha API
+ - InferenceModel is not depended upon by upstream or downstream components
+
+ Phase 1 will retain the Criticality functionality, but will rename the InferenceModel API and slim down the spec. Additionally, this slimmed down spec will be able to be applied at a _per request_ level. Justification in [Phase 1](#phase-1).
+
+### Phase 2 - Introduce new Usage Tracking, Fairness, & SLO CRDs
+Phase 2 will happen over a longer period of time & slowly introduce new CRDs to Inference Gateway, much of what is discussed in this proposal is keeping Phase 2 in mind, but phase 2 can be considered experimental & subject to change.
+
+Primarily phase 2 will introduce and develop these these CRDs:
+- Usage tracking (used in fairness)
+- Fairness configuration
+- SLO configuration
+
+## Design Principles
+
+### Goals
+- Reliable and predictable fairness allocation
+- Disconnect identity from policy-like objects where possible
+- Anonymous identity/defaults are graceful (fault-tolerant) & unsurprising
+- Scalable, simple, and reusable config
+- Retain the functionality of InferenceModel
+  - Traffic splitting models & modelName rewrite
+  - Criticality
+
+### Non-Goals
+- Addressing security concerns with the API, this is currently expected to either be:
+  - Entirely contained within a trusted system
+  - Or auth handled upstream
+- IGW implementing a custom auth mechanism
+
+
+## Definitions
+
+- **Tenant** Kuberenetes chooses the term ***tenant*** as described [here](https://kubernetes.io/docs/concepts/security/multi-tenancy/#tenants). Fairness APIs _may_ be used in multitenant scenarios, so as an example, multi-tenancy may be used.
+
+# Proposal
+
+Discussion of the problem(s) can be seen in the linked documents. Here we will describe the new API surface.
+
+## Phase 1
+
+### Structure change
+This API solves 3 general pillars of problem, that can also be categorized into 2 areas:
+
+Higher-order Request Gropuing (Usage tracking):
+ - This API describes Resource Sharing (Criticality/Fairness)
+ - This API describes Identification (used in Fairness)
+
+Request specific objectives:
+ - This API describes Specific Request Policy (SLO/Criticality)
+
+
+As such, the InferenceModel API will be split into separate CRDs to reflect the difference in these scopes. Phase 1 will focus on the **Request specific objectives**. Specifically it will maintain the inclusion of criticality. Other phase 1 changes:
+
+- The EPP will expose a flag to define the header key that will be used to assign InferenceObjectives to
```

Comment:
can we define the header key for both selecting the inferenceObjective and the flow id?